### DISCUSSION TITLE
uv venv not auto starting up when new terminal is selected

### DISCUSSION THREAD

**Discussion by WaldoJoubert-GH** [zed-industries/zed#18806] [Feature Requests]
I hope I'm not the only one struggling with this with zed on windows

"terminal": {
    "detect_venv": {
      "on": {
        // Default directories to search for virtual environments, relative
        // to the current working directory. We recommend overriding this
        // in your project's settings, rather than globally.
        "directories": [".venv", "venv"],
        // Can also be `csh`, `fish`, and `nushell`
        "activate_script": "power_shell"
      }
    }

**Comment by SirSilver** [comment]
For some reason it is not working for me until `.venv` dir is unfolded at least once in the project panel. But I use MacOS, so I'm not sure if it's related. Same for any venv, not only `uv`

**Comment by kurt-rhee** [comment]
Also not working for me using uv on macOS (activate_script: default)

**Comment by WaldoJoubert-GH** [comment]
i found that if i add my .venv to .gitignore then it wont run, dont know if this is by design.

**Comment by jacobwhall** [comment]
FYI #19227 is tracking this issue now.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
